hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/InternalSchemaUtils.java

```diff
@@ -57,15 +62,16 @@ public static InternalSchema pruneInternalSchema(InternalSchema schema, List<Str
     List<Integer> prunedIds = names.stream().map(name -> {
       int id = schema.findIdByName(name);
       if (id == -1) {
-        throw new IllegalArgumentException(String.format("cannot prune col: %s which does not exist in hudi table", name));
+        LOG.debug("Cannot find field name {} in schema {}, skipping.", name, schema);
+        return null;
       }
       return id;
-    }).collect(Collectors.toList());
+    }).filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
> cannot prune col: %s which does not exist in hudi table

The original logic makes more sense, what is the use case that to prune the schema with a non existing field?

Reply:
If the internal schema has less fields than the requested schema. In the writer path, the writer schema can have new columns that are not in the in file's existing schema.